packages/offix-datastore/src/storage/Storage.ts

```diff
@@ -1,57 +1,115 @@
+     * @param input The update to be made
+     * @param predicate A PredicateFunction to filter data
+     * @returns A Promise of the updated data
+     */
+    update(storeName: string, input: any, predicate?: PredicateFunction): Promise<any>;
+
+    /**
+     * Deletes data matching predicate or all from the store
+     *
+     * @param storeName The name of the store
+     * @param predicate A PredicateFunction to filter data
+     * @returns A Promise of the deleted data
+     */
+    remove(storeName: string, predicate?: PredicateFunction): Promise<any | any[]>;
 }
 
+/**
+ * The various change events that can occur on Store
+ *
+ * "ADD" - data was added to the Store
+ * "UPDATE" - data was updated in the Store
+ * "DELETE" - data was deleted from the Store
+ */
+export type EventTypes = "ADD" | "UPDATE" | "DELETE";
+
+/**
+ * StoreChangeEvent is an event emitted whenever
+ * a change has occurred on the local store
+*/
 export interface StoreChangeEvent {
-    operationType: string;
+    /**
+     * The type of change event that just occurred
+     */
+    eventType: EventTypes;
+
+    /**
+     * The data that was affected by the change
+     */
     data: any;
 }
 
 export class Storage {
     public readonly storeChangeEventStream: PushStream<StoreChangeEvent>;
     private adapter: IStorageAdapter;
 
-    constructor(models: Model[], schemaVersion: number) {
+    constructor(dbName: string, storeNames: string[], schemaVersion: number) {
         this.storeChangeEventStream = new ObservablePushStream();
-        this.adapter = createDefaultStorage(models, schemaVersion);
+        this.adapter = createDefaultStorage(dbName, storeNames, schemaVersion);
     }
 
-    async save(model: Model): Promise<PersistedModel> {
-        const persistedModel = { ...model, id: generateId() };
-        const result = await this.adapter.save(persistedModel);
+    public async save(storeName: string, input: any): Promise<any> {
+        const result = await this.adapter.save(storeName, { ...input, id: generateId() });
         this.storeChangeEventStream.push({
-            operationType: "ADD",
+            eventType: "ADD",
             data: result
         });
         return result;
     }
 
-    query(modelName: string, predicate?: PredicateFunction): Promise<PersistedModel | PersistedModel[]> {
-        return this.adapter.query(modelName, predicate);
+    public query(storeName: string, predicate?: PredicateFunction): Promise<any | any[]> {
+        return this.adapter.query(storeName, predicate);
     }
 
-    async update(model: PersistedModel): Promise<PersistedModel> {
-        const result = await this.adapter.update(model);
+    public async update(storeName: string, input: any, predicate?: PredicateFunction): Promise<any> {
+        const result = await this.adapter.update(storeName, input, predicate);
         this.storeChangeEventStream.push({
-            operationType: "UPDATE",
+            eventType: "UPDATE",
```

Comment:
I would love to have event types defined - typescript enum or even object with constants. WDYT?

Reply:
I thought [this](https://github.com/aerogear/offix/blob/55e7c25a13feabd37c7bb27b52f70ad62d522866/packages/offix-datastore/src/storage/Storage.ts#L60) was enough. We could use enums though.